### PR TITLE
Document the configuration and assignment of Host Devices

### PR DESCRIPTION
**Pull request by vladikr**
This PR provides documentation for Host Devices assignment in Kubevirt
https://github.com/kubevirt/kubevirt/pull/3937

Signed-off-by: Vladik Romanovsky <vromanso@redhat.com>

### CODE REVIEW

File: creation/host-devices.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+External providers:
+`externalResourceProvider` field indicates that this resource is being provided
+by an external device plugging. 
+KubeVirt in this case will only permit the usage of this device in the cluster
+but will leave the allocation and monitoring to an external device plugin.
+
+
+Statring a Virtual Machine
```
Comment (fabiand) [inline creation/host-devices.md:40]: Typo

File: creation/host-devices.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+This mechanism is generic and allows various types of PCI devices, such as GPU
+or any other devices attached to a PCI bus,  to be assigned. It also allows
+Mediated devices, such as pre-configured virtual GPUs to be assigned using the
+same mechanism.
+  
+
+Listing Permitted Devices
+-------------------------
+
+Administrators can control which host devices are permitted to be used in the
```
Comment (fabiand) [inline creation/host-devices.md:14]: are exposed and permitted to be used

File: creation/host-devices.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+cluster.
+Permitted host devices in the cluster will need to be whitelisted in KubeVirt
```
Comment (fabiand) [inline creation/host-devices.md:16]: s/whitelisted/allowlisted/

File: creation/host-devices.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+CR by its vendor:product selector for PCI devices or mediated device names
```
Comment (fabiand) [inline creation/host-devices.md:17]: `vendor:product`

File: creation/host-devices.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    - pciVendorSelector: "10DE:1EB8"
+      resourceName: "nvidia.com/TU104GL_Tesla_T4"
+      externalResourceProvider: true
+    - pciVendorSelector: "8086:6F54"
+      resourceName: "intel.com/QAT"
+    mediatedDevices:
+    - mdevNameSelector: "GRID T4-1Q"
+      resourceName: "nvidia.com/GRID_T4-1Q"
+```
+
```
Comment (fabiand) [inline creation/host-devices.md:32]: Please add small sections to explain how pciVendorSelector, mdevNameSelector works.

File: creation/host-devices.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+-------------------------
+
+Administrators can control which host devices are permitted to be used in the
+cluster.
+Permitted host devices in the cluster will need to be whitelisted in KubeVirt
+CR by its vendor:product selector for PCI devices or mediated device names 
+
+```
+configuration:
+  permittedHostDevices:
```
Comment (fabiand) [inline creation/host-devices.md:21]: s/permitted/exposed/?

File: creation/host-devices.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
+Host Devices Assignment
+=======================
+
+KubeVirt provides a mechanism for assigning host devices to a virtual machine.
+This mechanism is generic and allows various types of PCI devices, such as GPU
+or any other devices attached to a PCI bus,  to be assigned. It also allows
+Mediated devices, such as pre-configured virtual GPUs to be assigned using the
+same mechanism.
+  
+
```
Comment (fabiand) [inline creation/host-devices.md:10]: A note about setting the karg intel_iommu?

File: creation/host-devices.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
+Host Devices Assignment
+=======================
+
+KubeVirt provides a mechanism for assigning host devices to a virtual machine.
+This mechanism is generic and allows various types of PCI devices, such as GPU
+or any other devices attached to a PCI bus,  to be assigned. It also allows
```
Comment (fabiand) [inline creation/host-devices.md:6]: double space

File: creation/host-devices.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+Mediated devices, such as pre-configured virtual GPUs to be assigned using the
+same mechanism.
+
```
Comment (fabiand) [inline creation/host-devices.md:9]: Maybe a small note that today the slicing/pre-configured virtual GPUs needs to be done outside of kubevirt?

File: creation/host-devices.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+but will leave the allocation and monitoring to an external device plugin.
+
+
+Statring a Virtual Machine
+--------------------------
+HostDevices, as well as the existing GPUs field, will be able to reference both
+PCI and Mediated devices
+
+```
+spec:
```
Comment (fabiand) [inline creation/host-devices.md:46]: Please prepend:

kind: VirtualMachineInstance

**Review by fabiand — COMMENTED** [review]
A few notes

File: creation/host-devices.md (hunk 9 above)
Comment (vladikr) [reply creation/host-devices.md:9]: Done

File: creation/host-devices.md (hunk 7 above)
Comment (vladikr) [reply creation/host-devices.md:10]: Done

File: creation/host-devices.md (hunk 8 above)
Comment (vladikr) [reply creation/host-devices.md:6]: done

File: creation/host-devices.md (hunk 2 above)
Comment (vladikr) [reply creation/host-devices.md:14]: Done

File: creation/host-devices.md (hunk 3 above)
Comment (vladikr) [reply creation/host-devices.md:16]: Done

File: creation/host-devices.md (hunk 6 above)
Comment (vladikr) [reply creation/host-devices.md:21]: ?

Comment (vladikr) [reply creation/host-devices.md:21]: It's `permittedHostDevices` in the CR

File: creation/host-devices.md (hunk 10 above)
Comment (vladikr) [reply creation/host-devices.md:114]: Done

File: creation/host-devices.md (hunk 5 above)
Comment (vladikr) [reply creation/host-devices.md:87]: Done

**Review by fabiand — APPROVED** [review]

**Pull request merged**